src/construction.cpp

```diff
@@ -738,14 +733,23 @@ bool player_can_build( player &p, const inventory &inv, const std::string &desc
 
 bool player_can_build( player &p, const inventory &inv, const construction &con )
 {
+
     if( p.has_trait( trait_DEBUG_HS ) ) {
         return true;
     }
 
     if( !p.meets_skill_requirements( con ) ) {
         return false;
     }
-    return con.requirements->can_make_with_inventory( inv, is_crafting_component );
+
+    const bool can_build = con.requirements->can_make_with_inventory( inv, is_crafting_component );
+    if( !can_build ) {
+        popup( _( "You can't build that!" ) );
+    } else if( g->u.fine_detail_vision_mod() > 4 && !con.dark_craftable ) {
```

Comment:
This is referring to the player character, what's the purpose of the `player` parameter when the function is going to use the global player character anyway?